### PR TITLE
Initial Reference Server Implementation

### PR DESCRIPTION
**Pull request by jacmarjorie**
Opening this PR to get the initial server implementation merged into master. Includes:
- docker-compose spin up process
- simple importer
- test data (tcga live gene expression data)
- basic server implementation with all endpoints

Reviewing the import process now to validate other data import.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,376 @@
...
+1. `docker-compose build`
+
+1. `docker-compose up`
+
+1. To load test data, you need pandas. If you are in a virtualenv run `pip install pandas`. This is not included in server `requirements.txt` to decrease build and up time. Follow these commands to load: 
+
+```
+cd util
+python importer.py -c ../test/data/import.config -i ../test/data/tcgaLive*.tsv -H $(docker-machine ip default)
+```
```
Comment (bwalsh) [inline README.md:31]: Getting the following on first try ...

```
$ python importer.py -c ../test/data/import.config -i ../test/data/tcgaLive*.tsv -H $(docker-machine ip default)
Traceback (most recent call last):
  File "importer.py", line 7, in <module>
    from connector import Connector
  File "/Users/walsbr/framespace-ref/util/connector.py", line 1, in <module>
    import pymongo, os
ImportError: No module named pymongo
```

Comment (bwalsh) [inline README.md:31]: Needed to do `pip install pymongo`

Comment (bwalsh) [inline README.md:31]: Then I started to get 

```
$ python importer.py -c ../test/data/import.config -i ../test/data/tcgaLive*.tsv -H $(docker-machine ip default)
Traceback (most recent call last):
  File "importer.py", line 131, in <module>
    importer = Importer(args.config, args.inputs, host=args.host)
  File "importer.py", line 24, in __init__
    self.conn.registerAxes(self.config.axes, self.config.ksmajor_axis, self.config.ksminor_axis)
  File "/Users/walsbr/framespace-ref/util/connector.py", line 42, in registerAxes
    self.axis.update({'name': axis['name']}, axis, upsert=True)
  File "/Library/Python/2.7/site-packages/pymongo/collection.py", line 2235, in update
    with self._socket_for_writes() as sock_info:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/Library/Python/2.7/site-packages/pymongo/mongo_client.py", line 716, in _get_socket
    server = self._get_topology().select_server(selector)
  File "/Library/Python/2.7/site-packages/pymongo/topology.py", line 142, in select_server
    address))
  File "/Library/Python/2.7/site-packages/pymongo/topology.py", line 118, in select_servers
    self._error_message(selector))
pymongo.errors.ServerSelectionTimeoutError: 192.168.99.100:27017: [Errno 61] Connection refused
```

Comment (jacmarjorie) [inline README.md:31]: I forgot to mention the creation of the virtualenv, did you do a docker-compose build before docker-compose up?

Comment (jacmarjorie) [inline README.md:31]: @bwalsh can you try what I suggest above and let me know. I've got it working here, and I've updated the readme.

**Comment by bwalsh** [comment]
+1 : I followed the readme instructions, loaded data and exercised the example API calls.  everything worked as documented.

**Comment by jacmarjorie** [comment]
per @adamstruck 's suggestion need to add dataframe slice subsetting documentation

**Comment by jacmarjorie** [comment]
Initial import cleanup finished. Created a new task to address a more cleaned up import process. That will be next iteration. Merging this to master.

**Pull request merged**